blocks/extensions/saveload.py

```diff
@@ -0,0 +1,61 @@
+"""Extensions for saving and loading the state of a training process."""
+import dill
+
+from blocks.extensions import SimpleExtension
+
+
+class SaveLoadBase(SimpleExtension):
+    """The base class for save-load extensions.
+
+    Contains the logic that can be shared by different save-load
+    extensions.
+
+    """
+    def log_saving_done(self, destination):
+        """Makes a record in the log that saving has been done.
+
+        Parameters
+        ----------
+        destination : str
+            The destination where the state of the training process was
+            saved.
+
+        """
+        self.main_loop.log.current_row.saving_done_to = destination
+
+
+class DumpMainLoop(SaveLoadBase):
```

Comment:
Fair enough, but maybe that's partly because the interface is supposed to be `pickler = pickle.Pickler(f)` followed by `pickler.dump(object)` i.e. the `pickler` is an object that serializes and the `dump` method performs the writing. But given that `dump` "writes a pickled representation" and "pickling is alternatively known as 'serialization'" according to the docs, I think `SerializeMainLoop` works too, and is a bit clearer for me... I can live with "dump" though.